packages/eas-cli/src/metadata/upload.ts

```diff
@@ -37,32 +32,32 @@ export async function uploadMetadataAsync(
   Log.addNewLineIfNone();
   Log.log('Uploading App Store configuration...');
 
-  const errors: Error[] = [];
-  const config = createAppleReader(fileData);
-  const tasks = createAppleTasks(metadataCtx);
-  const taskCtx = { app };
-
-  for (const task of tasks) {
-    try {
-      await task.prepareAsync({ context: taskCtx });
-    } catch (error: any) {
-      errors.push(error);
+  await withTelemetryAsync(MetadataEvent.APPLE_METADATA_UPLOAD, { app, auth }, async () => {
+    const errors: Error[] = [];
+    const config = createAppleReader(fileData);
+    const tasks = createAppleTasks(metadataCtx);
+    const taskCtx = { app };
+
+    for (const task of tasks) {
+      try {
+        await task.prepareAsync({ context: taskCtx });
+      } catch (error: any) {
+        errors.push(error);
+      }
     }
-  }
 
-  for (const task of tasks) {
-    try {
-      await task.uploadAsync({ config, context: taskCtx as AppleData });
-    } catch (error: any) {
-      errors.push(error);
+    for (const task of tasks) {
+      try {
+        await task.uploadAsync({ config, context: taskCtx as AppleData });
+      } catch (error: any) {
+        errors.push(error);
+      }
     }
-  }
 
-  unsubscribeTelemetry();
-
-  if (errors.length > 0) {
-    throw new MetadataUploadError(errors, executionId);
-  }
+    if (errors.length > 0) {
+      throw new MetadataUploadError(errors);
+    }
+  });
```

Comment:
That's right, they are both the "task runners". The main difference is the task it's executing and the error it could be throwing. I can create a task runner, but I didn't want to abstract too much in the first version.

Reply:
Let's create a follow-up task to refactor this.